### ISSUE TITLE
[napari-workflows] Allow for workflow outputs (leafs) that aren't saved (not in output_specs dict)?

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-tasks-core#205]
The current napari workflow wrapper does some validation initially:
https://github.com/fractal-analytics-platform/fractal-tasks-core/blob/19393f16c01dd8e9d3a11942ae495a505ad52cb8/fractal_tasks_core/napari_workflows_wrapper.py#L93-L100

The check that all inputs are provided is important. The check that all workflow outputs are actually saved is too hard. I think we can relax this to a strong warning.

Reason: Users may have somewhat complex workflows that calculate things that aren't actually saved. This shouldn't be a typical use-case, as it doesn't make much sense to actually run the calculations then. But it doesn't break the workflow that we don't save all the outputs. Especially when users start sharing napari workflows, some users may run un-edited workflows, but are e.g. not interested in one of the measurement tables that get generated.

I'll test this a bit further to confirm, so no action needed just now. Just wanted to write it down so we don't forget about it.

**Comment by jluethi** [comment]
And an additional check we may want to think about: A user may want to save an output that is not a leaf (e.g. case 6 here, where the workflow generates labels & measurements: https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/31). In that case, the labels are not a leaf of the workflow, but should still be saved. I think that should work in the current implementation (currently setting up some examples to check it out), but we aren't checking whether the output names are valid names, right? We could check whether the output names appear as nodes of the dask graph. Because if they don't, then that input can't be saved.

**Comment by jluethi** [comment]
Ok, confirmed the issue with the hard output check. If I save the label images as output, but not the table generated as a leaf, I get the following error:
```
2022-11-15 13:46:43,373; INFO; Loaded workflow from /data/homes/jluethi/v1_fractal/fractal-demos/examples-v1/08_cardio_2x2_napari_workflows/tmp_cardio-2x2_multi_workflow_2/../np_wf_2_label.yaml
2022-11-15 13:46:43,374; ERROR; Some item of wf.leafs()=['regionprops_DAPI'] is not part of output_specs={'Result of Expand labels (scikit-image, nsbatwm)': {'type': 'label', 'label_name': 'wf_2_labels'}}.
Traceback (most recent call last):
  File "/net/nfs4/pelkmanslab-fileserver-common/data/homes/fractal/joel/fractal_v1/fractal-demos/examples-v1/server/FRACTAL_ROOT/.fractal/fractal-tasks-core0.3.3/venv/lib/python3.8/site-packages/fractal_tasks_core/napari_workflows_wrapper.py", line 571, in <module>
    run_fractal_task(
  File "/data/homes/fractal/joel/fractal_v1/fractal-demos/examples-v1/server/FRACTAL_ROOT/.fractal/fractal-tasks-core0.3.3/venv/lib/python3.8/site-packages/fractal_tasks_core/_utils.py", line 80, in run_fractal_task
    metadata_update = task_function(**task_args.dict())
  File "/net/nfs4/pelkmanslab-fileserver-common/data/homes/fractal/joel/fractal_v1/fractal-demos/examples-v1/server/FRACTAL_ROOT/.fractal/fractal-tasks-core0.3.3/venv/lib/python3.8/site-packages/fractal_tasks_core/napari_workflows_wrapper.py", line 96, in napari_workflows_wrapper
    raise ValueError(msg)
ValueError: Some item of wf.leafs()=['regionprops_DAPI'] is not part of output_specs={'Result of Expand labels (scikit-image, nsbatwm)': {'type': 'label', 'label_name': 'wf_2_labels'}}.
```

This should become a warning.

**Comment by tcompa** [comment]
I made it a warning with https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/209. It would be great if you could re-run the same test and then close this issue.

**Comment by jluethi** [comment]
ok, will test this once it's released :)

**Comment by tcompa** [comment]
Now available in 0.3.5.

**Comment by jluethi** [comment]
I tested this successfully with [0af4a83d9fd87b6640adf2e7ba638b805d4feeb8](https://github.com/fractal-analytics-platform/fractal-demos/commit/0af4a83d9fd87b6640adf2e7ba638b805d4feeb8)

We get the following warning in the log, as expected:
```
2022-11-21 09:55:54,720; WARNING; Some item of wf.leafs()=['regionprops_DAPI'] is not part of output_specs={'Result of Expand labels (scikit-image, nsbatwm)': {'type': 'label', 'label_name': 'wf_2_labels'}}.
```

Thus, this now works as intended. Thanks @tcompa !
